source/common/upstream/upstream_impl.cc

```diff
@@ -1382,6 +1383,23 @@ ClusterInfoImpl::upstreamHttpProtocol(absl::optional<Http::Protocol> downstream_
                                                                : Http::Protocol::Http11};
 }
 
+bool validateTransportSocketSupportsQuic(
+    const envoy::config::core::v3::TransportSocket& transport_socket) {
+  // The transport socket is valid for QUIC if it is either a QUIC transport socket,
+  // or if it is a QUIC transport socket wrapped in an HTTP/1.1 proxy socket.
+  if (transport_socket.name() == "envoy.transport_sockets.quic") {
+    return true;
+  }
+  if (transport_socket.name() != "envoy.transport_sockets.http_11_proxy" &&
+      transport_socket.name() != "envoy.transport_sockets.upstream_http_11_proxy") {
```

Comment:
@alyssawilk Looks like we also have an upstream version of the proxy transport socket.

Reply:
Where?  I see the one in source/extensions/extensions_metadata.yaml but I'm not seeing the latter in E-M or Envoy with a quick grep around
/wait-any

Reply:
I was seeing failures from test/extensions/transport_sockets/http_11_proxy/connect_integration_test.cc without this change. Maybe we should change the transport_socket name in that test?

Reply:
yeah, lets